src/tokio/seqpacket.rs

```diff
@@ -0,0 +1,200 @@
+    /// Connects to the socket named by path.
+    ///
+    /// This function will create a new Unix socket and connect to the path
+    /// specified, associating the returned stream with the default event loop's
+    /// handle.
+    pub async fn connect<P: AsRef<Path>>(path: P) -> io::Result<UnixSeqpacketConn> {
+        let conn = nonblocking::UnixSeqpacketConn::connect(path)?;
+        let conn = UnixSeqpacketConn::new(conn)?;
+
+        //poll_fn(|cx| conn.io.poll_write_ready(cx)).await?;
+        Ok(conn)
+    }
+
+    pub(crate) fn new(conn: nonblocking::UnixSeqpacketConn) -> io::Result<UnixSeqpacketConn> {
+        let io = PollEvented::new(conn)?;
+        Ok(UnixSeqpacketConn { io })
+    }
+}
+
+impl UnixSeqpacketConn {
+    /// Sends data on the socket to the socket's peer.
+    pub async fn send(&mut self, buf: &[u8]) -> io::Result<usize> {
+        poll_fn(|cx| self.poll_send_priv(cx, buf)).await
+    }
+
+    /// Receives data from the socket.
+    pub async fn recv(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        poll_fn(|cx| self.poll_recv_priv(cx, buf)).await
+    }
+
+    pub(crate) fn poll_recv_priv(
+        &self,
+        cx: &mut Context<'_>,
+        buf: &mut [u8],
+    ) -> Poll<io::Result<usize>> {
+        ready!(self.io.poll_read_ready(cx, mio::Ready::readable()))?;
+
+        match self.io.get_ref().recv(buf) {
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                self.io.clear_read_ready(cx, mio::Ready::readable())?;
+                Poll::Pending
+            }
+            Err(e) => Poll::Ready(Err(e)),
+            Ok((x, _truncated)) => Poll::Ready(Ok(x)),
+        }
+    }
+
+    pub(crate) fn poll_send_priv(
+        &self,
+        cx: &mut Context<'_>,
+        buf: &[u8],
+    ) -> Poll<io::Result<usize>> {
+        ready!(self.io.poll_write_ready(cx))?;
+
+        match self.io.get_ref().send(buf) {
+            Err(ref e) if e.kind() == io::ErrorKind::WouldBlock => {
+                self.io.clear_write_ready(cx)?;
+                Poll::Pending
+            }
+            x => Poll::Ready(x),
+        }
+    }
+}
+
+impl AsyncRead for UnixSeqpacketConn {
+    unsafe fn prepare_uninitialized_buffer(&self, _: &mut [MaybeUninit<u8>]) -> bool {
+        false
+    }
+
+    fn poll_read(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut [u8],
+    ) -> Poll<io::Result<usize>> {
+        self.poll_recv_priv(cx, buf)
+    }
+}
+
+impl AsyncWrite for UnixSeqpacketConn {
+    fn poll_write(
+        self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &[u8],
+    ) -> Poll<io::Result<usize>> {
+        self.poll_send_priv(cx, buf)
+    }
+
+    fn poll_flush(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        Poll::Ready(Ok(()))
+    }
+
+    fn poll_shutdown(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        // self.shutdown(std::net::Shutdown::Write)?;
```

Comment:
Ah, yes. Socket shutdown gets propagated to existing tokio/mio-uds sockets via, e.g. [`UnixStream::shutdown`](https://doc.rust-lang.org/nightly/std/os/unix/net/struct.UnixStream.html#method.shutdown), but a similar function doesn't yet exist for `UnixSeqpacketConn`. So that will need to get implemented first and then used to build this.

Reply:
Blocking/mio `shutdown()` done in 4d6c2c9, and tokio wrapper 6ed4903.

Reply:
Thanks